src/Subscriptions/Migrations/UpdateProductID.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Give\Subscriptions\Migrations;
+
+use Give\Donations\ValueObjects\DonationMetaKeys;
+use Give\Framework\Database\DB;
+use Give\Framework\Database\Exceptions\DatabaseQueryException;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Framework\Migrations\Exceptions\DatabaseMigrationException;
+
+/**
+ * @unreleased
+ *
+ * During the v2 form data transfer to the v3 form
+ * we update the _give_payment_form_id in donations meta table to reference the new v3 form id
+ * but that update was not reflected in give_subscriptions.product_id prior to this migration
+ *
+ * This migration will update all subscriptions that do not have the correct product_id value
+ */
+class UpdateProductID extends Migration
+{
+    /**
+     * @inheritDoc
+     */
+    public static function id(): string
+    {
+        return 'update_transferred_subscriptions_product_id';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function title(): string
+    {
+        return 'Update subscriptions to reflect the v2 form data transfer to the v3 form';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function timestamp(): string
+    {
+        return strtotime('2025-10-15 00:00:00');
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @throws DatabaseMigrationException
+     */
+    public function run()
+    {
+        try {
+            $query = <<<SQL
+                UPDATE %s AS subscriptions
+                INNER JOIN %s donationMeta
+                    ON donationMeta.meta_key = '%s'
+                    AND donationMeta.meta_value = subscriptions.id
+                INNER JOIN %s donationMeta2
+                    ON donationMeta2.donation_id = donationMeta.donation_id
+                    AND donationMeta2.meta_key = '%s'
+                SET subscriptions.product_id = donationMeta2.meta_value
+                WHERE subscriptions.product_id != donationMeta2.meta_value
+            SQL;
+
+            DB::query(sprintf(
+                $query,
+                DB::prefix('give_subscriptions'),
+                DB::prefix('give_donationmeta'),
+                DonationMetaKeys::SUBSCRIPTION_ID,
+                DB::prefix('give_donationmeta'),
+                DonationMetaKeys::FORM_ID,
```

Comment:
The tricky part here is that subscriptions have a one-to-many relationship with donations (renewals) so there can be _many_ donations with each subscription ID.  I think that will cause unnecessary joins.

Instead, I recommend only joining the initial donation for each subscription which can be achieved by using the parent_payment_id

Reply:
Resolved: https://github.com/impress-org/givewp/pull/8141/commits/1f077cb04497fd7f51eaedcc0a8e7dc40860a4a2